framework/include/base/NonlinearEigenSystem.h

```diff
@@ -62,6 +62,15 @@ class NonlinearEigenSystem
 
   virtual NumericVector<Number> & RHS() override;
 
+  // return the Nth converged eigenvlue
+  virtual std::pair<Real, Real>  getNthConvergedEigenvalue(dof_id_type n) { return _transient_sys.get_eigenpair(n); }
```

Comment:
Libmesh does not check the range, but slepc does. It is definitely a good idea to add the bounds checking in moose too.